enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,894 @@
+---
+title: set_eip_nlb_ingress
+authors:
+  - miheer
+reviewers:
+  - Miciah
+  - frobware
+  - gspence
+  - candace
+approvers:
+  - Miciah
+  - frobware
+  - gspence
+api-approvers:
+  - joel
+  - deads
+creation-date: 2024-03-06
+last-updated: 2024-05-13
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-1274"
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+# Set AWS EIP For NLB IngressController
+
+## Summary
+
+This enhancement allows a user to configure AWS Elastic IP(EIP) for an AWS Network Load Balancer(NLB) default or custom IngressController.
+This is a feature request to enhance the IngressController API to be able to support static IPs during
+- Custom NLB IngressController creation 
+- Reconfiguration of the custom and default router to NLB with EIP allocation.
+
+## Motivation
+
+### User Stories
+
+- As a cluster administrator, I want to provision EIPs and use them with an NLB IngressController.
+- As a cluster administrator, I want to reconfigure default router to use EIPs.
+- As a cluster administrator of OpenShift on AWS (ROSA), I want to use static IPs (and therefore AWS Elastic IPs) so that 
+  I can configure appropriate firewall rules.
+  I want the default AWS Load Balancer that they use (NLB) for their router to use these EIPs.
+- As a cluster administrator who has purchased a block of public IPv4 addresses, I want to use these IP addresses, so 
+  that I can avoid Amazon's new charge for public IP addresses.
+
+### Goals
+- Users are able to use EIP for a NLB `IngressController`.
+- The load balancer of an existing `IngressController` can be reconfigured to use specific EIPs by updating the `IngressController` specification and recreating the load balancer as type NLB.
+
+### Non-Goals
+
+- Creation of EIPs in AWS.
+- Monitoring or management of changes to EIPs in the AWS environment.
+- Static IP usage with NLBs for OpenShift API server.
+- Check the number of public subnets available in AWS environment and then compare number of subnets
+  with the number of EIPs the user provides. Note - If subnets are provided in the IngressController CR
+  we do compare the subnets with the number of eipAllocations provided in the IngressController CR. 
```

Comment:
I would move this note up to the Goal section:
```
The number of subnets provided in the default IngressController specification will be validated to ensure it matches the number of eipAllocations.
```

Reply:
Also, the RFE is asking for an installation option, so you should point that out as a **non-goal** of this enhancement since it was moved out of this enhancement.